### DISCUSSION TITLE
[Esame 9 Settembre 2019] Esercizio 3

### DISCUSSION THREAD

**Discussion by Elia-Belli** [sapienzastudentsnetwork/calcolo-delle-probabilita#228] [Esami - Bertini @ Informatica]
![immagine](https://github.com/sapienzastudentsnetwork/calcprob2324/assets/107944476/1585959a-5fa2-408b-bee5-5b623b4f2db0)

> Concettualmente simile a:
- #175

**Comment by LeoCosta02** [comment]
![image](https://github.com/sapienzastudentsnetwork/calcprob2324/assets/128373356/90a35de1-385f-46e7-b0f3-6da9f3a83fbc)

**Comment by Elia-Belli** [comment]
**(1)** Nel caso peggiore l'algoritmo richiede $n$ confronti, quindi $C_n=1,...,n$

**(2)** La distribuzione **NON è geometrica**: una v.a geometrica indica il numero di prove indipendenti prima di ottenere un successo con probabilità **p** ad ogni prova; nel nostro caso ad ogni prova la probabilità di trovare la stringa varia, infatti se non troviamo la stringa al primo confronto passa da $\frac{1}{n}$ a $\frac{1}{n-1}$ al secondo confronto.

Mostriamo la probabilità per alcuni valori, ricordando che la probabilità di trovare la stringa è il prodotto delle probabilità di trovarla a questo confronto per la probabilità di non averla trovata nei precedenti:

$$\mathbb{P}(C_n=1)=\frac{1}{n}$$

$$\mathbb{P}(C_n=2)=\mathbb{P}((trovata\ al\ 2°)\cap (non\ trovata\ al\ 1°))=\frac{1}{n-1}\cdot \frac{n-1}{n}=\frac{1}{n}$$

$$\mathbb{P}(C_n=3)=\mathbb{P}((trovata\ al\ 3°)\cap (non\ trovata\ al\ 1°\ e\ 2°))=\frac{1}{n-2}\cdot \frac{n-1}{n}\cdot \frac{n-2}{n-1}=\frac{1}{n}$$

> Si tratta di un prodotto telescopico

$$\mathbb{P}(C_n=k)= \mathbb{P}((trovata\ al\ k°)\cap(non\ trovata\ al\ 1°,2°,...,k-1°))=\frac{1}{n}$$

**(3)** Utilizziamo la definizione di valore atteso:

$$\mathbb{E}(C_n)=\sum\_{k=1}^nk\cdot \mathbb{P}(C_n=k)=\sum\_{k=1}^n \frac{k}{n}=\frac{1}{n}\cdot \frac{n\cdot (n+1)}{2}=\frac{n+1}{2}$$
